### PR TITLE
boards: xtensa: add Kincony KC868-A32 module suppport

### LINKED ISSUE #67321: tests/lib/devicetree/devices fails on KC868-A32 board
**Describe the bug**
tests/lib/devicetree/devices fails on Kincony KC868-A32 board

**To Reproduce**
1. west build -b kincony_kc868_a32  tests/lib/devicetree/devices/

**Expected behavior**
Test pass

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/7172682929/job/19546571568

**Environment (please complete the following information):**
 - OS: Linux
 - Zephyr SDK 0.16.4,
 - ver 3.5.99

### PR DESCRIPTION
**Pull request by bbilas**
Add support for the KC868-A32 ESP32 home automation relay module.

### CODE REVIEW

File: tests/lib/devicetree/devices/testcase.yaml [hunk 1]
```diff
@@ -18,3 +18,4 @@ tests:
       - mimxrt595_evk_cm33
       - nrf9131ek_nrf9131
       - nrf9131ek_nrf9131_ns
+      - kincony_kc868_a32
```
Comment (henrikbrixandersen) [inline tests/lib/devicetree/devices/testcase.yaml:21]: Why is this added?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: tests/lib/devicetree/devices/testcase.yaml (hunk 1 above)
Comment (bbilas) [reply tests/lib/devicetree/devices/testcase.yaml:21]: https://github.com/zephyrproject-rtos/zephyr/actions/runs/7172682929/job/19546571568

File: boards/xtensa/kincony_kc868_a32/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+Enabling Ethernet
+*****************
+
+Enable Ethernet in KConfig:
+
+.. code-block:: cfg
+
+    CONFIG_NETWORKING=y
+    CONFIG_NET_L2_ETHERNET=y
```
Comment (sylvioalves) [inline boards/xtensa/kincony_kc868_a32/doc/index.rst:91]: Would you include a link to the board? https://www.kincony.com/arduino-esp32-32-channel-relay-module-kc868-a32.html

File: boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 4 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 5 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-names = "default";
+
+	pcf8574@21 {
```
Comment (sylvioalves) [inline boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts:95]: Both I2C0 and I2C1 contains the same IO expander. Is it correct?

Comment (bbilas) [reply boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts:95]: There are 8 identical expanders on the board, 4 for inputs and 4 for outputs. They have the same i2c address but are connected to different buses.

File: tests/lib/devicetree/devices/testcase.yaml (hunk 1 above)
Comment (sylvioalves) [reply tests/lib/devicetree/devices/testcase.yaml:21]: For some reason, `CONFIG_I2C` is forced to be y when test case sets to n, breaking it. That is weird.

File: boards/xtensa/kincony_kc868_a32/doc/index.rst (hunk 2 above)
Comment (bbilas) [reply boards/xtensa/kincony_kc868_a32/doc/index.rst:91]: Done.

File: boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts (hunk 3 above)
Comment (bbilas) [reply boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts:95]: @sylvioalves I was wondering if I should give them unique node labels as they cannot be used using the i2c shell at the moment as the device list returns duplicate device IDs.

<pre>
uart:~$ device list
devices:
- pcf8574@25 (READY)
- pcf8574@24 (READY)
- pcf8574@22 (READY)
- pcf8574@21 (READY)
- pcf8574@25 (READY)
- pcf8574@24 (READY)
- pcf8574@22 (READY)
- pcf8574@21 (READY)
</pre>

We could also improve this by adding an i2c bus prefix to the identifier, for example:

<pre>
uart:~$ device list
devices:
- i2c0_pcf8574@25 (READY)
- i2c0_pcf8574@24 (READY)
- i2c0_pcf8574@22 (READY)
- i2c0_pcf8574@21 (READY)
- i2c1_pcf8574@25 (READY)
- i2c1_pcf8574@24 (READY)
- i2c1_pcf8574@22 (READY)
- i2c1_pcf8574@21 (READY)
</pre>

File: tests/lib/devicetree/devices/testcase.yaml (hunk 1 above)
Comment (bbilas) [reply tests/lib/devicetree/devices/testcase.yaml:21]: Good point, I'll try to check that too.

File: boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts (hunk 3 above)
Comment (sylvioalves) [reply boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts:95]: That is interesting. I guess a new PR enabling such prefix is the way.

Comment (bbilas) [reply boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts:95]: This can be done in this PR, we just need to do something like this:

<pre>
    i2c0_pcf8574@21 {
         compatible = "nxp,pcf8574";
         reg = <0x21>;
         gpio-controller;
         #gpio-cells = <2>;
         ngpios = <8>;
    };
</pre>

Comment (sylvioalves) [reply boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts:95]: ahá, of course. When I read it before I though you wanted to add the peripheral prefix in every and all instances.  
Sure, it makes sense.

Comment (bbilas) [reply boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts]: Done.

File: tests/lib/devicetree/devices/testcase.yaml (hunk 1 above)
Comment (bbilas) [reply tests/lib/devicetree/devices/testcase.yaml:21]: @henrikbrixandersen @sylvioalves I've done a quick investigation and didn't find anything that can be related to it. Can we fix this later as it is a merge blocker?

Comment (henrikbrixandersen) [reply tests/lib/devicetree/devices/testcase.yaml:21]: I think we should try to find the root cause of this. 

@nashif Any ideas?

Comment (bbilas) [reply tests/lib/devicetree/devices/testcase.yaml:21]: @henrikbrixandersen Since nobody has looked at it, can we put it aside?

Comment (henrikbrixandersen) [reply tests/lib/devicetree/devices/testcase.yaml:21]: @bbilas Yes. Please open a separate issue for tracking this.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@mbolivar-ampere Please review the change to the devicetree test exclusion.
